src/github/githubRepository.ts

```diff
@@ -206,7 +209,29 @@ export class GitHubRepository implements IGitHubRepository, vscode.Disposable {
 	}
 
 	async getPullRequests(prType: PRType, page?: number): Promise<PullRequestData | undefined> {
-		return prType === PRType.All ? this.getAllPullRequests(page) : this.getPullRequestsForCategory(prType, page);
+		return await (prType === PRType.All ? this.getAllPullRequests(page) : this.getPullRequestsForCategory(prType, page));
+	}
+
+	public async ensureRepositoryReturnsAvatar(testAvatarUrl: string): Promise<boolean> {
+		if(this._repositoryReturnsAvatar === null) {
+			let response: AxiosResponse | null = null;
+
+			try {
+				response  = await axois({method: 'get', url: testAvatarUrl, maxRedirects: 0});
+			} catch (err) {
+				if(err && err instanceof Error) {
+					response = (<any> err).response as AxiosResponse;
+				}
+			}
+
+			if(response && response.status === 200) {
+				this._repositoryReturnsAvatar = true;
+			}
+
+			this._repositoryReturnsAvatar = false;

```

Comment:
I think we're missing an `else` here, or a return in the 200 check

Reply:
Thanks, I'll send a PR to fix this.

Reply:
https://github.com/Microsoft/vscode-pull-request-github/pull/1145